packages/next/next-server/server/image-optimizer.ts

```diff
@@ -44,30 +43,31 @@ export async function imageOptimizer(
     return { finished: true }
   }
 
-  let absoluteUrl: URL
-  try {
-    absoluteUrl = new URL(url)
-  } catch (_error) {
-    // url was not absolute so assuming relative url
+  let absoluteUrl: URL | undefined
+  let relativeUrl: string | undefined
+
+  if (url.startsWith('/')) {
+    relativeUrl = url
+  } else {
     try {
-      absoluteUrl = new URL(url, `${proto}://${host}`)
-    } catch (__error) {
+      absoluteUrl = new URL(url)
+    } catch (_error) {
       res.statusCode = 400
       res.end('"url" parameter is invalid')
       return { finished: true }
     }
-  }
 
-  if (!['http:', 'https:'].includes(absoluteUrl.protocol)) {
-    res.statusCode = 400
-    res.end('"url" parameter is invalid')
-    return { finished: true }
-  }
+    if (!['http:', 'https:'].includes(absoluteUrl.protocol)) {
+      res.statusCode = 400
+      res.end('"url" parameter is invalid')
+      return { finished: true }
+    }
 
-  if (!server.renderOpts.dev && !domains.includes(absoluteUrl.hostname)) {
-    res.statusCode = 400
-    res.end('"url" parameter is not allowed')
-    return { finished: true }
+    if (!server.renderOpts.dev && !domains.includes(absoluteUrl.hostname)) {
```

Comment:
```suggestion
    if (!domains.includes(absoluteUrl.hostname)) {
```

This no longer needs to be excluded for `next dev`. These two validations for protocol and hostname should only run for absolute, not relative urls.